test/util/bitcoin-util-test.py

```diff
@@ -63,87 +67,93 @@ def bctester(testDir, input_basename, buildenv):
     else:
         sys.exit(0)
 
-def bctest(testDir, testObj, buildenv):
+
+def bctest(test_dir, test_obj, build_env):
     """Runs a single test, comparing output and RC to expected output and RC.
 
     Raises an error if input can't be read, executable fails, or output/RC
     are not as expected. Error is caught by bctester() and reported.
     """
     # Get the exec names and arguments
-    execprog = buildenv["BUILDDIR"] + "/src/" + testObj['exec'] + buildenv["EXEEXT"]
-    execargs = testObj['args']
-    execrun = [execprog] + execargs
+    exec_prog = os.path.join(build_env["BUILDDIR"], "src", test_obj["exec"]) + build_env["EXEEXT"]
+    exec_args = test_obj["args"]
+    exec_run = [exec_prog] + exec_args
 
     # Read the input data (if there is any)
-    stdinCfg = None
-    inputData = None
-    if "input" in testObj:
-        filename = testDir + "/" + testObj['input']
-        inputData = open(filename).read()
-        stdinCfg = subprocess.PIPE
+    stdin_cfg = None
+    input_data = None
+    if "input" in test_obj:
+        filename = os.path.join(test_dir, test_obj["input"])
+        input_data = open(filename).read()
+        stdin_cfg = subprocess.PIPE
 
     # Read the expected output data (if there is any)
-    outputFn = None
-    outputData = None
-    if "output_cmp" in testObj:
-        outputFn = testObj['output_cmp']
-        outputType = os.path.splitext(outputFn)[1][1:]  # output type from file extension (determines how to compare)
+    output_fn = None
+    output_data = None
+    output_type = None
+    if "output_cmp" in test_obj:
+        output_fn = test_obj["output_cmp"]
+        output_type = os.path.splitext(output_fn)[1][1:]  # output type from file extension (determines how to compare)
         try:
-            outputData = open(testDir + "/" + outputFn).read()
+            output_data = open(os.path.join(test_dir, output_fn)).read()
         except:
-            logging.error("Output file " + outputFn + " can not be opened")
+            logging.error("Output file " + output_fn + " can not be opened")
             raise
-        if not outputData:
-            logging.error("Output data missing for " + outputFn)
+        if not output_data:
+            logging.error("Output data missing for " + output_fn)
+            raise Exception
+        if not output_type:
```

Comment:
Generally, I'd prefer if refactoring is separate (at least a separate commit) from changing behavior.